eureka-client/src/main/java/com/netflix/discovery/shared/transport/jersey/JerseyEurekaHttpClientFactory.java

```diff
@@ -64,9 +71,10 @@ public static EurekaHttpClientFactory create(EurekaClientConfig clientConfig,
                 .withMaxTotalConnections(clientConfig.getEurekaServerTotalConnections())
                 .withConnectionIdleTimeout(clientConfig.getEurekaConnectionIdleTimeoutSeconds())
                 .withEncoder(clientConfig.getEncoderName())
-                .withDecoder(clientConfig.getDecoderName(), clientConfig.getClientDataAccept());
+                .withDecoder(clientConfig.getDecoderName(), clientConfig.getClientDataAccept())
+                .withClientIdentity(clientIdentity);
 
-        if (isSecure && "true".equals(System.getProperty("com.netflix.eureka.shouldSSLConnectionsUseSystemSocketFactory"))) {
+        if ("true".equals(System.getProperty("com.netflix.eureka.shouldSSLConnectionsUseSystemSocketFactory"))) {
             clientBuilder.withClientName("DiscoveryClient-HTTPClient-System").withSystemSSLConfiguration();
```

Comment:
Maybe we should only set base name, and append protocol specific parts in the builder? So all those `if`s can be replaced with single `clientBuilder.withClientName("DiscoveryClient")`?